### PR TITLE
Fix execution icons based on executed icons

### LINKED ISSUE #1990: Running part of workflow should not care about the rest of the workflow
If I try to run a part of the workflow (or even just one tool), then Spine Engine checks the whole workflow first. There are probably good reasons for this, but there are currently some unintended consequences. These are things that are in other parts of the workflow and shouldn't influence running the selected part:
- It asks whether I want to upgrade all DBs in the workflow (if they need upgrading)
- If I use the new 'filter validation' option in the scenario arrow, then the whole workflow will not run if there are no scenarios. This is pretty bad: I can't fill an empty DB with an importer, because there are no scenarios (even though I'm not running that part of the workflow)
-  It can be very slow sometimes to check parts of the worfklow (that are not run). I don't know what it is, but this happens for an result importer item in the FlexTool workflow (again, I'm not executing it, it's just being checked - maybe looking what inputs in what scenarios are available or something).

### PR DESCRIPTION
**Pull request by PekkaSavolainen**
Execution icons on Design View are cleared for items that are not part of the current DAG executed by Spine Engine.

Fixes #1990

## Checklist before merging
- [ ] Documentation is up-to-date
- [ ] Release notes have been updated
- [ ] Unit tests have been added/updated accordingly
- [ ] Code has been formatted by black
- [ ] Unit tests pass

### CODE REVIEW

File: spinetoolbox/project_item_icon.py [hunk 1]
```diff
@@ -590,6 +590,9 @@ def __init__(self, parent):
     def item_name(self):
         return self._parent.name()
 
+    def _hide_execution_icon(self):
+        self.hide()
```
Comment (soininen) [inline spinetoolbox/project_item_icon.py:594]: Makes me wonder if we really need this method? Why not just call `hide()`?

Comment (PekkaSavolainen) [reply spinetoolbox/project_item_icon.py]: Thanks, the method is gone now.

**Comment by soininen** [comment]
I just came to my mind that `CHANGELOG.md` could be updated to mention that users will need to select Data Transformers for execution after these changes.

**Comment by jkiviluo** [comment]
Refreshing this. Would be very nice to get this in.

**Pull request merged**